cg/apps/sequencing_metrics_parser/parsers/bcl_convert.py

```diff
@@ -0,0 +1,204 @@
+        parsed_metrics: List[BclConvertQualityMetrics] = []
+        metrics_reader = ReadFile.get_content_from_file(
+            file_format=FileFormat.CSV, file_path=self.quality_metrics_path, read_to_dict=True
+        )
+        for row in metrics_reader:
+            parsed_metrics.append(
+                BclConvertQualityMetrics(
+                    lane=int(row["Lane"]),
+                    sample_internal_id=row["SampleID"],
+                    read_pair_number=row["ReadNumber"],
+                    yield_bases=int(row["Yield"]),
+                    yield_q30_bases=int(row["YieldQ30"]),
+                    quality_score_sum=int(row["QualityScoreSum"]),
+                    mean_quality_score_q30=float(row["Mean Quality Score (PF)"]),
+                    q30_bases_percent=float(row["% Q30"]),
+                )
+            )
+        return parsed_metrics
+
+    def parse_demux_metrics_file(self) -> List[BclConvertDemuxMetrics]:
+        """Reads a BCL convert demux metrics file into the BclConvertDemuxMetrics model."""
+        LOG.info(f"Parsing BCLConvert metrics file: {self.demux_metrics_path}")
+        parsed_metrics = []
+        metrics_reader = ReadFile.get_content_from_file(
+            file_format=FileFormat.CSV, file_path=self.demux_metrics_path, read_to_dict=True
+        )
+        for row in metrics_reader:
+            parsed_metrics.append(
+                BclConvertDemuxMetrics(
+                    lane=int(row["Lane"]),
+                    sample_internal_id=row["SampleID"],
+                    sample_project=row["Sample_Project"],
+                    read_pair_count=row["# Reads"],
+                    perfect_index_reads_count=row["# Perfect Index Reads"],
+                    perfect_index_reads_percent=row["% Perfect Index Reads"],
+                    one_mismatch_index_reads_count=row["# One Mismatch Index Reads"],
+                    two_mismatch_index_reads_count=row["# Two Mismatch Index Reads"],
+                )
+            )
+            return parsed_metrics
+
+    def parse_adapter_metrics_file(self) -> List[BclConvertAdapterMetrics]:
+        LOG.info(f"Parsing BCL convert adapter metrics file {self.adapter_metrics_path}")
+        parsed_metrics: List[BclConvertAdapterMetrics] = []
+        metrics_reader = ReadFile.get_content_from_file(
+            file_format=FileFormat.CSV, file_path=self.adapter_metrics_path, read_to_dict=True
+        )
+        for row in metrics_reader:
+            parsed_metrics.append(
+                BclConvertAdapterMetrics(
+                    lane=int(row["Lane"]),
+                    sample_internal_id=row["Sample_ID"],
+                    sample_project=row["Sample_Project"],
+                    read_number=row["ReadNumber"],
+                    sample_bases=row["SampleBases"],
+                )
+            )
+        return parsed_metrics
+
+    def get_nr_of_header_lines_in_sample_sheet(
```

Comment:
Seems like this should be a general function that can be reused elsewhere as well and not anchored to this object

Reply:
You are right. I will leave this for now here and will make a new PR for a BclConvert sample sheet reader with validators, readers etc. once that is done I will replace